### PR TITLE
Add diamond.execute.contractAddress option

### LINKED ISSUE #233: Diamond initialization function need to be in a facet
In `HardhatRuntimeEnvironment.deployments.diamond.deploy`, you can only specify an initialization routine that exists in one of the facets. This, however, is not foreseen by the [EIP-2535 standard](https://eips.ethereum.org/EIPS/eip-2535):

```solidity
    /// @notice Add/replace/remove any number of functions and optionally execute
    ///         a function with delegatecall
    /// @param _diamondCut Contains the facet addresses and function selectors
    /// @param _init The address of the contract or facet to execute _calldata
    /// @param _calldata A function call, including function selector and arguments
    ///                  _calldata is executed with delegatecall on _init
    function diamondCut(
        FacetCut[] calldata _diamondCut,
        address _init,
        bytes calldata _calldata
    ) external;
```

A diamond cut is able to later call *any* contract and pass *any* calldata. This should be reflected in the `HardhatRuntimeEnvironment.deployments.diamond.deploy` function because it forces developers to create a new facet just for initialization and take extra precautions measures to avoid reinitializations (such as storing a boolean flag in storage, which is unnecessary gas wasted by the deploying agent).

### PR DESCRIPTION
**Pull request by guidanoli**
Closes #233

### CODE REVIEW

**Comment by wighawag** [comment]
Thanks for the PR. unfortunately, this change is not enough, as it won't work on first deployment.
I am planning to change the Diamantaire or even get rid of it and then allow to pass a different contract address for execution

File: src/helpers.ts [hunk 1]
```diff
@@ -1943,11 +1943,20 @@ Note that in this case, the contract deployment will not behave the same if depl
 
     let data = '0x';
     if (options.execute) {
-      const diamondContract = new Contract(
+      let executeContractABI;
+      if (options.execute.contractAddress) {
+        const artifact = await partialExtension.getExtendedArtifact(
+            options.execute.contractAddress
```
Comment (wighawag) [inline src/helpers.ts:1949]: getExtendedArtifact expect a name, not an address

Comment (guidanoli) [reply src/helpers.ts:1949]: You are right, maybe we should accept a `Contract` object instead?

Comment (wighawag) [reply src/helpers.ts:1949]: or a name

Comment (guidanoli) [reply src/helpers.ts:1949]: Right, right.

**Pull request closed without merging**